big_o/big_o.py

```diff
@@ -173,3 +173,52 @@ def big_o(func, data_generator,
         fitted['times'] = time
 
     return best, fitted
+
+def print_report(best, others):
+    """ Prints the detailed summary of the calculated reports
+
+    Input:
+    ------
+
+    best -- Object representing the complexity class that best fits
+            the measured execution times.
+
+    others -- A dictionary of fitted complexity classes to the residuals
+
+    Output:
+    -------
+
+    None
+
+    """
+
+    def print_line(c_rank, name, coef, residual):
+        print('{:>5}  |  {:12}  |  {:35}  |  {:.2G}'.format(
+            c_rank, name, coef, residual))
+
+    def parse_data(obj):
+        name = '{}'.format(obj.__class__.__name__)
+        if obj.coeff is None:
+            coef = 'not yet fitted'
+        else:
+            coef = obj.format_str().format(*obj.coefficients()).replace('time',
+                                                                        '').replace('=', '')+' (sec)'
```

Comment:
The fact that you need to replace `time = ` with empty strings is a smell that something is strange... in `complexities`, please remove `time = ` from all the `format_str` functions, and have `__str__` pre-pend it.

Reply:
Sorted